File: recursion/cpp/recursion_pure.cpp
#include <iostream>

using namespace std;

int factorial(int n) {
    if (n == 1) return 1;
    return n * factorial(n-1);
}

void solve(int test_cases) {
    if (test_cases == 0) return;

    int n; cin >> n;
    cout << factorial(n) << endl;
    solve(test_cases-1);
}

int main() {
    int test_cases; cin >> test_cases;
    solve(test_cases);
}

File: recursion/python/recursion_pure.py
def factorial(n):
    if (n == 1): return 1
    return n * factorial(n-1)


def solve(test_cases):
    if (test_cases == 0): return
    n = int(input())
    print(factorial(n))
    solve(test_cases-1)


def main():
    test_cases = int(input())
    solve(test_cases)


# Expected:
# 1
# 720 -> 6 * 5 * 4 * 3 * 2 * 1
# 5040

if __name__ == "__main__":
    main()